### DISCUSSION TITLE
On-premise deployment for production environments

### DISCUSSION THREAD

**Discussion by ryzalk** [flyteorg/flyte#1373] [Q&A] [answered]
Hi there! Are there any recommendations and/or guides for deploying Flyte on on-premise clusters? The only available guide I can find relevant to this is the sandbox deployment but I understand that is not for production.

**Comment by kumare3** [comment, marked as answer]
Hi @ryzalk so there are 3 major dependencies for Flyte
1. K8s
2. Postgres
3. S3 or some blob store. As part of flyte v0.17.0 we are adding support for fsspec and / or bring your own blob store. Natively flyte supports s3, gcs, etc

**Reply by kumare3** [reply]
please join the slack channel and we can help. But, if you do not have S3 or something equivalent that is supported by [Stow](https://github.com/graymeta/stow) then it will probably not work. An example could be minio

**Reply by ryzalk** [reply]
Thanks for the reply @kumare3 . I will join the Slack chat and converse more there then.
